### ISSUE TITLE
'NSLocationWhenInUseUsageDescription' is needed even if no location is in use

### ISSUE THREAD

**Issue by DennisOstertag** [react-native-maps/react-native-maps#4166]
# Bug report

## Environment info

      react-native: 0.64.2
      react: 17.0.2
      react-native-maps: 0.30.1

`react-native info` output:

```System:
    OS: Windows 10 10.0.19042
    CPU: (12) x64 Intel(R) Core(TM) i7-10850H CPU @ 2.70GHz
    Memory: 14.11 GB / 31.61 GB
  Binaries:
    Node: 16.1.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 7.11.2 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK: Not Found
  IDEs:
    Android Studio: Version     2020.3.0.0 AI-203.7717.56.2031.7583922
    Visual Studio: 16.11.32126.315 (Visual Studio Professional 2019), 17.1.32210.238 (Visual Studio Professional 2022)
  Languages:
    Java: 11.0.11 - C:\Program Files\Common Files\Oracle\Java\javapath\javac.EXE
  npmPackages:
    @react-native-community/cli: Not Found
    react: ~17.0.2 => 17.0.2
    react-native: ~0.64.2 => 0.64.2
    react-native-windows: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

## Steps to reproduce

- You must provide a clear list of steps and code to reproduce the problem.
- Keep the code reproducing the bug as simple as possible, with the minimum amount of code required to reproduce the issue. See https://stackoverflow.com/help/mcve.
- If this library has additional install steps, describe them (e.g., pod install? jetify? etc).
- Either re-create the bug using the repository's example app or link to a GitHub repository with code that reproduces the bug.
- Explain the steps we need to take to reproduce the issue:

1. Install latest version of react native maps
2. Use MapView and set the property `showsUserLocation={false}`
3. Do NOT add the NSLocationWhenInUseUsageDescription to the Info.plist
4. Upload the app to apple

Describe what you expected to happen:

As there is no access to the location, the app should pass Apples review and the 'NSLocationWhenInUseUsageDescription' should not be needed.

Describe what you actually happens:

The app gets rejected by apple with the issue:
```
ITMS-90683: Missing Purpose String in Info.plist. ... App should contain a NSLocationWhenInUseUsageDescription key.
```

## Same issue but closed without a fix:
#3305 

There is a suggested fix, that works, but i need to patch react-native-maps, so it is not the best way to do.
_Originally posted by @edanedison in https://github.com/react-native-maps/react-native-maps/issues/3305#issuecomment-668745054_

**Comment by Simon-TechForm** [comment]
I support this change being implemented. The same thing applies to android, as `react-native-maps` adds the ACCESS_COARSE_LOCATION permission to the android manifest, which in best case is redundant and in worst case causes conflicts.

**Comment by meliodev** [comment]
Any solution or workaround for this issue ?

**Comment by DennisOstertag** [comment]
> Any solution or workaround for this issue ?

yes as i posted above you can patch react native maps and remove the line that requests permission.

_Originally posted by @edanedison in https://github.com/react-native-maps/react-native-maps/issues/3305#issuecomment-668745054_

[Link to patch package](https://anu-thomas.medium.com/patch-package-in-react-native-c7786a15e279)

**Comment by ewfian** [comment]
Can we add some flag to conditional compile

**Comment by DennisOstertag** [comment]
> Hi guys, I have the same issue but for the play store. As a patch, I tried to remove the line `<uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />` in `android/src/main/AndroidManifest.xml` But it didn't change anything 😢 any idea for a quick or long-term fix?

hi there,
andorid is quite simple.
```   
 <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" tools:node="remove" />
 <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" tools:node="remove" />
```

node tools will remove these permissions for the whole app, this might not be the most pretty way to deal with it, but it should work

**Comment by LeandreArnaud** [comment]
Thanks @DennisOstertag !
It didn't worked but I finally added :
`
    "android": {
      "permissions": [],
`
In my `app.json` and it fixed my problem 🆒

**Comment by llapa** [comment]
I tried the @DennisOstertag suggested patch-package fix, but Apple still continues to nag about this permission. I am building a Expo app (eas build), where patch-packages should work. I can't upgrade to the beta-releases either, since Expo does not support those (app is crashing with beta -releases)

I also tried to patch with the specific commits, that were introduced in 1.0.0-beta.1. Same issue persists.

**Comment by ewfian** [comment]
It still there if Google Map API enabled.

**Comment by DennisOstertag** [comment]
> It still there if Google Map API enabled.

Thats weird, we are using google maps on android and apple maps for ios. We managed to get the app in both stores without any location permissions displayed/used.
@ewfian did the `tools:node="remove"` work for you when you add it to the permission in the android manifest?

**Comment by llapa** [comment]
@ewfian Yes , it just occured to me that in my use case, I use google as the provider for iOS also. Is there any way to get around this issue, if google is the provider? I do not actually use user location in any shape or form, I have also set the 'showUserLocation' flag to false.

**Comment by ewfian** [comment]
> @ewfian Yes , it just occured to me that in my use case, I use google as the provider for iOS also. Is there any way to get around this issue, if google is the provider? I do not actually use user location in any shape or form, I have also set the 'showUserLocation' flag to false.

@llapa I found a lot of references about Location and CoreLocation in Google Maps for iOS. I think this is the point.

**Comment by llapa** [comment]
So all suggested fixes (included in beta) won't work for iOS, if you use Google as the provider for the map? Is there any way around this?

**Comment by Simon-TechForm** [comment]
As @ewfian correcly pointed out, I'm afraid there's no way around providing `NSLocationWhenInUseUsageDescription` when using Google Maps on iOS. Google included code to request location permissions directly in the SDK, which flags the app in the Apple app review. I will make sure to include this in the iOS installation guide.
